### PR TITLE
Fix CloudWatch Insights Addon - User passed version not being respected

### PR DESCRIPTION
**Pull request by 5herlocked**
Added test cases to ensure all information is being passed downstream.

*Issue #, if available:* #948

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: lib/addons/cloud-watch-insights/index.ts [hunk 1]
```diff
@@ -75,11 +67,12 @@ export class CloudWatchInsights extends CoreAddOn {
       insightsSA.role.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('CloudWatchAgentServerPolicy'));
```
Comment (elamaran11) [inline lib/addons/cloud-watch-insights/index.ts:67]: Why are we not using `createNamespace` override method ?

File: lib/addons/cloud-watch-insights/index.ts [hunk 2]
```diff
@@ -75,11 +67,12 @@ export class CloudWatchInsights extends CoreAddOn {
...
       insightsSA.role.addManagedPolicy(ManagedPolicy.fromAwsManagedPolicyName('AWSXrayWriteOnlyAccess'));
```
Comment (elamaran11) [inline lib/addons/cloud-watch-insights/index.ts:68]: why are we not using `createSA` override method of `coreAddon` ?

File: lib/addons/cloud-watch-insights/index.ts [hunk 3]
```diff
@@ -75,11 +67,12 @@ export class CloudWatchInsights extends CoreAddOn {
...
 
-      const insightsAddon = new eks.CfnAddon(context.scope,  "CloudWatchInsightsAddon", {
+      const insightsAddon = new eks.CfnAddon(clusterInfo.cluster.stack,  "CloudWatchInsightsAddon", {
         addonName: defaultProps.addOnName,
```
Comment (elamaran11) [inline lib/addons/cloud-watch-insights/index.ts:71]: lets not do this. Lets use the `deploy` method of the `coreAddon` so we dont lose the functionality of `deploy` method of `coreAddon` which has features like `provideVersion` etc. With this approach if `auto` is passed, `auto` is passed as version to addon creation.

File: lib/addons/cloud-watch-insights/index.ts [hunk 4]
```diff
@@ -39,7 +32,6 @@ export type CloudWatchInsightsAddOnProps = Omit<CoreAddOnProps, "saName" | "addO
 const defaultProps = {
   addOnName: "amazon-cloudwatch-observability",
   version: "auto",
-  versionMap: versionMap,
   saName: "cloudwatch-agent",
```
Comment (elamaran11) [inline lib/addons/cloud-watch-insights/index.ts:35]: Where are we using `merge` to merge default props with passed props?

**Review by elamaran11 — CHANGES_REQUESTED** [review]
@5herlocked Thankyou for the rapid quick PR. Some feedback, lets not completely override the `deploy` method with your implementation, we are completely missing all functionalities of `deploy` method on `coreAddon.` What we have to do is, make sure merge of props is taken care, `createNamespace` and `createSA` methods are used via ovveride. Most importantly make sure following test cases are tested locally:
1. When `auto` is passed for `version`, it should make aws sdk call to grab the version.
2. When `auto` is passed and you terminal wont have access to aws, during building time, make sure the versionMap is used.
3. Actual version is passed.

**Comment by 5herlocked** [comment]
Ack.

Will submit an update to this tomorrow.

**Comment by shapirov103** [comment]
@5herlocked what is the ETA to address review feedback?

**Comment by 5herlocked** [comment]
ETA for reworked code is 03/20

**Comment by 5herlocked** [comment]
Rebased from main before committing, I apologize, will fix commit history.

File: lib/addons/cloud-watch-insights/index.ts [hunk 5]
```diff
@@ -1,12 +1,11 @@
...
+import {ClusterInfo, Values} from "../../spi";
+import {conflictsWith, supportsALL} from "../../utils";
 import {CoreAddOn, CoreAddOnProps} from "../core-addon";
 import {ebsCollectorPolicy} from "./iam-policy";
-import {ManagedPolicy} from "aws-cdk-lib/aws-iam";
-import { KubernetesVersion } from "aws-cdk-lib/aws-eks";
+import {IManagedPolicy, ManagedPolicy} from "aws-cdk-lib/aws-iam";
+import {KubernetesVersion, ServiceAccount} from "aws-cdk-lib/aws-eks";
 
 const versionMap: Map<KubernetesVersion, string> = new Map([
```
Comment (shapirov103) [inline lib/addons/cloud-watch-insights/index.ts:10]: Is this version map valid for 1.29 and previous releases? if not, let's fix it.

**Review by shapirov103 — COMMENTED** [review]
Looks much better, ty. One small comment, I will kick off e2e after that.

**Review by elamaran11 — COMMENTED** [review]
@5herlocked Approach looks good to me. Did you test all the below scenarios or should you update the tests for below :
1. `auto` with sdk call fetching version
2. `auto` with `versionMap` providing the version (Terminal having no access to AWS during build time).
3. Actual version passed
4. `auto` passed and no access to AWS and K8s version missing in `versionMap`. Error sceario.

File: lib/addons/cloud-watch-insights/index.ts (hunk 5 above)
Comment (5herlocked) [reply lib/addons/cloud-watch-insights/index.ts:10]: It is valid, though I am going to upgrade the default to v1.4 for the newer ones

**Comment by 5herlocked** [comment]
Tested with and without credentials in the environment. Feel free to e2e

**Review by elamaran11 — COMMENTED** [review]
LGTM

**Review by elamaran11 — APPROVED** [review]
LG to merge.

**Pull request merged**